src/pykeen/metrics/ranking.py

```diff
@@ -74,12 +79,107 @@ def expected_value(
         Compute expected metric value.
 
         Prefers analytical solution, if available, but falls back to numeric estimation via summation,
-        cf. `numeric_expected_value`.
+        cf. :func:`numeric_expected_value`.
         """
         if num_samples is None:
             raise ValueError("Numeric estimation requires to specify a number of samples.")
         return self.numeric_expected_value(num_candidates=num_candidates, num_samples=num_samples)
 
+    def numeric_variance(
+        self,
+        num_candidates: np.ndarray,
+        num_samples: int,
+    ) -> float:
+        """Compute variance by summation."""
+        num_candidates = np.asarray(num_candidates)
+        generator = np.random.default_rng()
+        ranks = generator.integers(low=1, high=num_candidates + 1, size=(num_samples, *num_candidates.shape))
+        return np.apply_along_axis(self, 0, ranks).var().item()
+
+    def variance(
+        self,
+        num_candidates: np.ndarray,
+        num_samples: Optional[int] = None,
+    ) -> float:
+        """Compute variance.
+
+        Prefers analytical solution, if available, but falls back to numeric estimation via summation,
+        cf. :func:`numeric_variance`.
+        """
+        if num_samples is None:
+            raise ValueError("Numeric estimation requires to specify a number of samples.")
+        return self.numeric_variance(num_candidates=num_candidates, num_samples=num_samples)
+
+
+class IncreasingZMixin(RankBasedMetric):
+    """A mixin to create a z-scored metric."""
+
+    value_range = ValueRange(lower=None, upper=None)
+
+    def __call__(self, ranks: np.ndarray, num_candidates: Optional[np.ndarray] = None) -> float:  # noqa: D102
+        v = super().expected_value(num_candidates=num_candidates) - super().__call__(
+            ranks=ranks, num_candidates=num_candidates
+        )
+        variance = super().variance(num_candidates=num_candidates)
+        return v / max(math.sqrt(variance), EPSILON)
```

Comment:
extract utility `save_divide` for `x / max(y, EPSILON)`?

Reply:
6d6a7ec3